### ISSUE TITLE
Padding is nonzero

### ISSUE THREAD

**Issue by taseppa** [fiorix/go-diameter#38]
When examining pcap files with Wireshark I see that i quite often get a warning saying AVP is malformed because padding is zero. I have noticed this issue only quite recently so could it be that some recent change is responsible for this?

**Comment by jaroszan** [comment]
Can you please specify AVP type for which it is happening?

**Comment by taseppa** [comment]
I noticed the problem with Session-Id (UTF8String) and Origin-Host (DiameterIdentity) at least.

**Comment by jaroszan** [comment]
I can see it as well, however for nested AVPs UTF8String is padded correctly in my case.

**Comment by jaroszan** [comment]
In my test case I have the following AVPs padded incorrectly:
- Session-Id
- Destination-Realm
- Service-Context-Id

**Comment by jaroszan** [comment]
Curiously when I run examples scripts (client/server) all AVPs are padded correctly.

**Comment by jaroszan** [comment]
It seems that issue is reproducible when client's bench option is set to true in examples/client/client.go. AVPs in ACR/ACA are padded incorrectly.

**Comment by jaroszan** [comment]
It looks that it is somehow dependent on the order in which AVPs are added to a message.

**Comment by fiorix** [comment]
This is not a fix for it, still trying to reproduce. I believe this bug was recently introduced by one of the changes in the encoder.

**Comment by fiorix** [comment]
Can someone send me a pcap with the error? Can't reproduce using -bench=true, ACR/ACA looks ok here.

**Comment by fiorix** [comment]
Never mind, I can reproduce it.

**Comment by fiorix** [comment]
To get to the bottom of this one I instrumented the message and avp encoders with hex.Dump and noticed both were actually correct. However, when encoding messages we (re)use buffers from a sync.Pool, and those buffers were eventually dirty from previous usage. So the change is quite simple: reset padding bytes after serializing the avp. Ended up updating the serialization logic to be more straightforward, should be good now.
